### PR TITLE
Geometry, GeometryCollision Tests

### PR DESCRIPTION
**Pull request by at669**
## Proposed change(s)

Introduces tests for runtime classes UrdfGeometry and UrdfGeometryCollision.

### Useful links (GitHub issues, JIRA tickets, forum threads, etc.)

[AIRO-671](https://jira.unity3d.com/browse/AIRO-671)

### Types of change(s)

- [ ] Bug fix
- [ ] New feature
- [ ] Code refactor
- [ ] Documentation update
- [x] Other (please describe): Unit tests

## Testing and Verification

Creating Unit tests.

### Test Configuration:
- Unity Version: 2020.2.0b9
- Unity machine OS + version: MacOS 10.15.7

## Checklist
- [x] Ensured this PR is up-to-date with the `dev` branch
- [x] Created this PR to target the `dev` branch
- [x] Followed the style guidelines as described in the [Contribution Guidelines](../CONTRIBUTING.md)
- [x] Added tests that prove my fix is effective or that my feature works
- [ ] Updated the documentation as appropriate

### CODE REVIEW

File: com.unity.robotics.urdf-importer/Tests/Runtime/Extensions/UrdfGeometryTests.cs [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        [SetUp]
+        public void SetUp()
+        {
+            box = new Link.Geometry(box: new Link.Geometry.Box(new double[] { 2, 3, 4 }));
+            cylinder = new Link.Geometry(cylinder: new Link.Geometry.Cylinder(1, 4));
+            sphere = new Link.Geometry(sphere: new Link.Geometry.Sphere(1));
+        }
+
+        [Test]
+        public void ExportGeometryData_Box()
```
Comment (peifeng-unity) [inline com.unity.robotics.urdf-importer/Tests/Runtime/Extensions/UrdfGeometryTests.cs:27]: naming convention?

File: com.unity.robotics.urdf-importer/Tests/Runtime/Extensions/UrdfGeometryTests.cs [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        [Test]
+        public void ExportGeometryData_Cylinder()
+        {
+            // Force runtime mode to avoid creating asset file
+            RuntimeURDF.runtimeModeEnabled = true;
+            var parent = new GameObject("Parent").transform;
+            UrdfGeometryCollision.Create(parent, GeometryTypes.Cylinder);
+
+            var t = parent.transform.Find("Cylinder");
+            Assert.IsNotNull(UrdfGeometry.ExportGeometryData(GeometryTypes.Cylinder, t));
```
Comment (peifeng-unity) [inline com.unity.robotics.urdf-importer/Tests/Runtime/Extensions/UrdfGeometryTests.cs:44]: Might be better if you could check if the exported geometry is equal to a `new Link.Geometry`; similar to the `ExporterGeometryData_Sphere()`

File: com.unity.robotics.urdf-importer/Tests/Runtime/Extensions/UrdfGeometryTests.cs [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+
+        [Test]
+        public void IsTransformed_Rotation()
+        {
+            var parent = new GameObject("Parent").transform;
+            parent.transform.localRotation = Quaternion.Euler(0, 30, 0);
+            Assert.IsTrue(UrdfGeometry.IsTransformed(parent, UrdfGeometry.GetGeometryType(box)));
+        }
+    }
+}
```
Comment (peifeng-unity) [inline com.unity.robotics.urdf-importer/Tests/Runtime/Extensions/UrdfGeometryTests.cs:119]: add a tailing line

**Comment by peifeng-unity** [comment]
In addition, would you mind increasing [the criteria of the test coverage in Yamato](https://github.com/Unity-Technologies/URDF-Importer/blob/dev/.yamato/yamato-config.yml#L18), e.g. increase it to 6% like`if [[ $linecoverage < 6 ]]; then exit 1; fi`

File: com.unity.robotics.urdf-importer/Tests/Runtime/Extensions/UrdfGeometryCollisionTests.cs [hunk 4]
```diff
@@ -18,6 +18,17 @@ public void SetUp()
             RuntimeURDF.AssetDatabase_CreateFolder("Assets/Tests", "Runtime");
             RuntimeURDF.AssetDatabase_CreateFolder("Assets/Tests/Runtime", "GeometryTests");
         }
+
+        [Test]
+        public void Create_Box_UnityPrimitive()
```
[the remark this answers is not in the document]
Comment (at669) [reply com.unity.robotics.urdf-importer/Tests/Runtime/Extensions/UrdfGeometryCollisionTests.cs:23]: i'm not 100% sure here, i was basing it on this [Best Practices](https://docs.microsoft.com/en-us/dotnet/core/testing/unit-testing-best-practices#best-practices) page, but i may be missing something!

File: com.unity.robotics.urdf-importer/Tests/Runtime/Extensions/UrdfGeometryCollisionTests.cs [hunk 5]
```diff
@@ -31,10 +42,33 @@ public void Create_CylinderMesh_AssetCreated()
             
             // Verify Cylinder created in Hierarchy
             Assert.IsNotNull(GameObject.Find("Cylinder").activeInHierarchy);
+            Assert.IsNotNull(GameObject.Find("Cylinder").GetComponent<MeshCollider>());
```
[the remark this answers is not in the document]
Comment (at669) [reply com.unity.robotics.urdf-importer/Tests/Runtime/Extensions/UrdfGeometryCollisionTests.cs:45]: the default cylinder collider is just a [capsule collider](https://docs.unity3d.com/Manual/PrimitiveObjects.html), so to be physically more accurate a new cylinder-shaped mesh is generated i believe

File: com.unity.robotics.urdf-importer/Tests/Runtime/Extensions/UrdfGeometryCollisionTests.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (peifeng-unity) [reply com.unity.robotics.urdf-importer/Tests/Runtime/Extensions/UrdfGeometryCollisionTests.cs:23]: Ah, I see. I was following the naming convention from the perception package like [this example](https://github.com/Unity-Technologies/com.unity.perception/blob/master/com.unity.perception/Tests/Runtime/Randomization/ScenarioTests/ScenarioTests.cs). It seems that the ML-Agent uses [the same naming convention](https://github.com/Unity-Technologies/ml-agents/blob/main/com.unity.ml-agents/Tests/Runtime/Utils/TestClasses.cs). Maybe we should use the same as theirs?

File: com.unity.robotics.urdf-importer/Tests/Runtime/Extensions/UrdfGeometryCollisionTests.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (peifeng-unity) [reply com.unity.robotics.urdf-importer/Tests/Runtime/Extensions/UrdfGeometryCollisionTests.cs:45]: Ah, okay, get you!

File: com.unity.robotics.urdf-importer/Tests/Runtime/Extensions/UrdfGeometryCollisionTests.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (at669) [reply com.unity.robotics.urdf-importer/Tests/Runtime/Extensions/UrdfGeometryCollisionTests.cs:23]: just discovered our internal [Testing](https://confluence.unity3d.com/display/AIRO/Testing) conventions, which calls out `MethodName_Scenario_ExpectedBehaviour()`!

**Review by peifeng-unity — DISMISSED** [review]

**Comment by peifeng-unity** [comment]
Looks like there are a few methods that are not tested in the [Geometry](https://yamato-artifactviewer.prd.cds.internal.unity3d.com/aed0a771-a4eb-4e35-b569-25020579d5b9%2Flogs%2Ftest-results%2FReport/Unity.Robotics.URDFImporter_UrdfGeometry.html) and [GeometryCollision](https://yamato-artifactviewer.prd.cds.internal.unity3d.com/aed0a771-a4eb-4e35-b569-25020579d5b9%2Flogs%2Ftest-results%2FReport/Unity.Robotics.URDFImporter_UrdfGeometryCollision.html). Are you planning to add them in the future PRs?

**Comment by at669** [comment]
> @peifeng-unity: Looks like there are a few methods that are not tested in the [Geometry](https://yamato-artifactviewer.prd.cds.internal.unity3d.com/aed0a771-a4eb-4e35-b569-25020579d5b9%2Flogs%2Ftest-results%2FReport/Unity.Robotics.URDFImporter_UrdfGeometry.html) and [GeometryCollision](https://yamato-artifactviewer.prd.cds.internal.unity3d.com/aed0a771-a4eb-4e35-b569-25020579d5b9%2Flogs%2Ftest-results%2FReport/Unity.Robotics.URDFImporter_UrdfGeometryCollision.html). Are you planning to add them in the future PRs?

most of the ones I didn't cover revolve around importing a mesh, with a valid info from an XML node or from file--personally thought it might fit more with integration testing or down-the-line testing since this doesn't currently include URDF files, but i am open to suggestions!

**Review by peifeng-unity — APPROVED** [review]

**Review by peifeng-unity — APPROVED** [review]

**Pull request merged**